src/misc/inc/QuadratureBase.h

```diff
@@ -0,0 +1,60 @@
+//-----------------------------------------------------------------------bl-
+//--------------------------------------------------------------------------
+//
+// QUESO - a library to support the Quantification of Uncertainty
+// for Estimation, Simulation and Optimization
+//
+// Copyright (C) 2008-2017 The PECOS Development Team
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the Version 2.1 GNU Lesser General
+// Public License as published by the Free Software Foundation.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc. 51 Franklin Street, Fifth Floor,
+// Boston, MA  02110-1301  USA
+//
+//-----------------------------------------------------------------------el-
+
+#ifndef UQ_QUADRATURE_BASE_H
+#define UQ_QUADRATURE_BASE_H
+
+#include <vector>
+
+#include <queso/asserts.h>
+
+namespace QUESO
+{
+  //! Base class for quadrature rules
+  /*! This serves as the base class for approximating integrals using numerical quadrature. Namely,
+   *   \f[ \int_{\Omega} f(x) \; dx \approx \sum_{i=1}^N f(x_i) w_i \f]
+   * This class contains the weights. Subclasses will have the positions since the position can be one-dimensional
+   * (Base1DQuadrature) or multi-dimensional (MultiDQuadratureBase).
+   */
+  class QuadratureBase
```

Comment:
Conceptually, it is abstract, but there are no virtual functions to put here. The subclass constructors will populate the weights (in this class) and the positions (in the first layer of subclasses, either 1D or multiD). Basically, this is avoiding putting weights in two different places.

We could do without 1DQuadrature entirely, and adhere to the interface in MultiDQuadratureBase, but since 1DQuadrature was already there, and its interface was specifically 1D, I went the route that I did.